src/traces/bar/defaults.js

```diff
@@ -33,7 +155,23 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     coerce('base');
     coerce('offset');
     coerce('width');
-    coerce('text');
+
+    coerceArray('text', coerceString);
+
+    var textPosition = coerceArray('textposition', coerceEnumerated);
+
+    var hasBoth = Array.isArray(textPosition) || textPosition === 'auto',
+        hasInside = hasBoth || textPosition === 'inside',
+        hasOutside = hasBoth || textPosition === 'outside';
+    if(hasInside || hasOutside) {
+        var textFont = coerceArray('textfont', coerceFont, layout.font);
+        if(hasInside) {
+            coerceArray('insidetextfont', coerceFont, textFont, layout.font);
+        }
+        if(hasOutside) {
+            coerceArray('outsidetextfont', coerceFont, textFont, layout.font);
+        }
+    }
```

Comment:
It looks like you've based this on [pie/defaults](https://github.com/plotly/plotly.js/blob/master/src/traces/pie/defaults.js#L51-L62) but you've got more logic around arrays in the input. So two questions:
- for the cases that *are* covered by pie, does this give the same inheritance from `layout.font` and `textfont` to `insidetextfont` and `outsidetextfont`, particularly around cases where parts of the font come from one default and other parts come from the other one? for example if `layout.font` supplies the family, `textfont` supplies the size, and `insidetextfont` makes it white.
- what extra cases does this cover and should we extend pie to cover these too?